index.js

```diff
@@ -38,114 +47,117 @@ function BlindsHTTPAccessory(log, config) {
-
     // the target position (0-100%)
     // https://github.com/KhaosT/HAP-NodeJS/blob/master/lib/gen/HomeKitTypes.js#L1564
     this.service
         .getCharacteristic(Characteristic.TargetPosition)
         .on('get', this.getTargetPosition.bind(this))
         .on('set', this.setTargetPosition.bind(this));
+
+    this.service
+        .getCharacteristic(Characteristic.PositionState)
+        .updateValue(Characteristic.PositionState.STOPPED);
 }
 
 BlindsHTTPAccessory.prototype.getCurrentPosition = function(callback) {
-    this.log("Requested CurrentPosition: %s", this.lastPosition);
+    if (this.verbose) {
+        this.log(`Requested CurrentPosition: ${this.lastPosition}%`);
+    }
     callback(null, this.lastPosition);
 }
 
-BlindsHTTPAccessory.prototype.getPositionState = function(callback) {
-    this.log("Requested PositionState: %s", this.currentPositionState);
-    callback(null, this.currentPositionState);
-}
-
 BlindsHTTPAccessory.prototype.getTargetPosition = function(callback) {
-    this.log("Requested TargetPosition: %s", this.currentTargetPosition);
+    if (this.verbose) {
+        this.log(`Requested TargetPosition: ${this.currentTargetPosition}%`);
+    }
     callback(null, this.currentTargetPosition);
 }
 
 BlindsHTTPAccessory.prototype.setTargetPosition = function(pos, callback) {
-    this.log("Set TargetPosition: %s", pos);
+    if (this.lagTimeout != null) clearTimeout(this.lagTimeout);
+    if (this.stopTimeout != null) clearTimeout(this.stopTimeout);
+    if (this.stepInterval != null) clearInterval(this.stepInterval);
+
     this.currentTargetPosition = pos;
     if (this.currentTargetPosition == this.lastPosition) {
-        if (this.interval != null) clearInterval(this.interval);
-        if (this.timeout != null) clearTimeout(this.timeout);
-        this.log("Already here");
-        callback(null);
-        return;
+        if (this.currentTargetPosition % 100 > 0) {
+            this.log(`Already there: ${this.currentTargetPosition}%`);
+            callback(null);
+            return;
+        } else {
+            this.log(`Already there: ${this.currentTargetPosition}%, re-sending request`);
+        }
     }
-    const moveUp = (this.currentTargetPosition >= this.lastPosition);
-    this.log((moveUp ? "Moving up" : "Moving down"));
 
-    this.service
-        .setCharacteristic(Characteristic.PositionState, (moveUp ? 1 : 0));
-
-    this.httpRequest((moveUp ? this.upURL : this.downURL), this.httpMethod, function() {
-        this.log(
-            "Success moving %s",
-            (moveUp ? "up (to " + pos + ")" : "down (to " + pos + ")")
-        );
-        this.service
-            .setCharacteristic(Characteristic.CurrentPosition, pos);
-        this.service
-            .setCharacteristic(Characteristic.PositionState, 2);
-    }.bind(this));
+    const moveUp = (this.currentTargetPosition > this.lastPosition || this.currentTargetPosition == 1);
```

Comment:
@zwerch sure.

Suppose `this.lastPosition` = 1 and user requests `this.currentTargetPosition  = 1` state. Without this logic, blinds would go down because moveUp = false.

Before, the way the logic was written, it would just abort without sending any commands if this.lastPosition = this.currentTargetPosition. But, this caused problems for me if a command was sent again for example. So this change was part of that change.